src/services/SelfTest.ts

```diff
@@ -85,70 +110,107 @@ export default class SelfTest {
           ws.on('error', (err: any) => {
             log.error('Error on connection for websockets', err)
             this.config.selfTestSuccess = false
-            process.exit(1)
+            this.wsSuccess = false
           })
         })
 
         const serverPromise = new Promise(async resolve => {
           try {
             const serverRes = await axios.get(`https://${response.manifestHash}.${url.host}`)
             const serverCheck = serverRes.data
-            log.debug('Pod upload succeeded', serverCheck)
+            log.debug('Pod HTTP request succeeded', serverCheck)
             if (serverCheck.imageUploaded) {
               // resolve promise
-              this.uploadSuccess = true
+              this.httpSuccess = true
               resolve()
             }
           } catch (err) {
-            log.error('Test contract not running', err)
+            log.error('Test pod not reachable', err)
             this.config.selfTestSuccess = false
-            process.exit(1)
+            this.httpSuccess = false
```

Comment:
Shouldn't this cause the promise to reject instead of allowing it to hang infinitely?